src/main/java/io/unlogged/AgentCommandExecutorImpl.java

```diff
@@ -483,23 +483,104 @@ public AgentCommandResponse injectMocks(AgentCommandRequest agentCommandRequest)
 
         for (String sourceClassName : mocksBySourceClass.keySet()) {
             Object sourceClassInstance = logger.getObjectByClassName(sourceClassName);
+			
+			Class<? extends Object> classObject = null;
             if (sourceClassInstance == null) {
-                // no instance found for this class
-                // nothing to inject mocks to
-                continue;
+
+				// define applicationContext and springBeanFactory
+				if (this.springTestContextManager == null) {
+					this.springTestContextManager = logger.getObjectByClassName("org.springframework.boot.web" +
+							".reactive.context" +
+							".AnnotationConfigReactiveWebServerApplicationContext");
+					springBeanFactory = setSpringApplicationContextAndLoadBeanFactory(this.springTestContextManager);
+				}
+	
+				if (this.springTestContextManager == null) {
+					this.springTestContextManager = logger.getObjectByClassName(
+							"org.springframework.boot.web.servlet.context.AnnotationConfigServletWebServerApplicationContext"
+					);
+					springBeanFactory = setSpringApplicationContextAndLoadBeanFactory(this.springTestContextManager);
+				}
+
+				if (applicationContext != null) {
+					// get list of bean names
+					// reflection: String[] beanNames = applicationContext.getBeanDefinitionNames()
+					Class<?> applicationContextClass = Class.forName("org.springframework.context.ApplicationContext");
+					Method getBeanDefinitionNamesMethod = applicationContextClass.getMethod("getBeanDefinitionNames");
+					String[] beanNames = (String[]) getBeanDefinitionNamesMethod.invoke(applicationContext);
+
+					// get bean from beanName
+					List<Object> applicationBean = new ArrayList<>();
+					for (String beanName: beanNames) {
+						// reflection: Object bean = applicationContext.getBean(beanName)
+
+						Method getBeanMethod = applicationContextClass.getMethod("getBean", String.class);
+						Object bean = getBeanMethod.invoke(applicationContext, beanName);
+						applicationBean.add(bean);
+					}
+
+					// inject mocks
+					// TODO: improve search performance here
+					Class<? extends Object> classDefination = null;
+					for (Object bean: applicationBean) {
```

Comment:
we have the applicationBean at this point. why now lookup its class and then lookup its constructor and create the bean (instance) again ? so the object now which is created in the flow that follows its not the instance which spring is going to use to serve requests. which means no mocks.

Reply:
ok, got it. I have changed the flow to use the object that spring is using. I still need to do the search to match the correct bean, but have now removed the logic where constructors were getting invoked.